docs/source/ssl_approaches/barlow_twins.rst

```diff
@@ -0,0 +1,134 @@
+
+.. code-block:: bash
+
+    python tools/run_distributed_engines.py config=pretrain/barlow_twins/barlow_twins_4node_resnet
+
+
+Using Synchronized BatchNorm for training
+--------------------------------------------
+
+For training Barlow Twins models, we convert all the BatchNorm layers to Global BatchNorm. For this, VISSL supports `PyTorch SyncBatchNorm <https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html>`_
+module and `NVIDIA's Apex SyncBatchNorm <https://nvidia.github.io/apex/_modules/apex/parallel/optimized_sync_batchnorm.html>`_ layers. Set the config params :code:`MODEL.SYNC_BN_CONFIG.SYNC_BN_TYPE` to :code:`apex` or :code:`pytorch`.
+
+If you want to use Apex, VISSL provides :code:`anaconda` and :code:`pip` packages of Apex (compiled with Optimzed C++ extensions/CUDA kernels). The Apex
+packages are provided for all versions of :code:`CUDA (9.2, 10.0, 10.1, 10.2, 11.0), PyTorch >= 1.4 and Python >=3.6 and <=3.9`.
+
+To use SyncBN during training, one needs to set the following parameters in configuration file:
+
+.. code-block:: yaml
+
+    MODEL:
+      SYNC_BN_CONFIG:
+        CONVERT_BN_TO_SYNC_BN: True
+        SYNC_BN_TYPE: apex
+        # 1) if group_size=-1 -> use the VISSL default setting. We synchronize within a
+        #     machine and hence will set group_size=num_gpus per node. This gives the best
+        #     speedup.
+        # 2) if group_size>0 -> will set group_size=value set by user.
+        # 3) if group_size=0 -> no groups are created and process_group=None. This means
+        #     global sync is done.
+        GROUP_SIZE: 8
+
+
+Vary the training loss settings
+------------------------------------------------
+Users can adjust several settings from command line to train the model with different hyperparams. For example: to use a different
+lambda of 0.01:
+
+.. code-block:: bash
+
+    python tools/run_distributed_engines.py config=pretrain/barlow_twins/barlow_twins_4node_resnet \
+        config.LOSS.barlow_twins_loss.lambda_=0.01
+
+The full set of loss params that VISSL allows modifying:
+
+.. code-block:: yaml
+
+    barlow_twins_loss:
+      lambda_: 0.01
+      scale_loss: 0.024
+      embedding_dim: 8192
+
+
+Training different model architecture
+------------------------------------------------
+VISSL supports many backbone architectures including ResNe(X)ts, wider ResNets. Some examples below:
+
+
+* **Train ResNet-101:**
+
+.. code-block:: bash
+
+    python tools/run_distributed_engines.py config=pretrain/barlow_twins/barlow_twins_4node_resnet \
+        config.MODEL.TRUNK.NAME=resnet config.MODEL.TRUNK.TRUNK_PARAMS.RESNETS.DEPTH=101
+
+* **Train ResNet-50-w2 (2x wider):**
+
+.. code-block:: bash
+
+    python tools/run_distributed_engines.py config=pretrain/barlow_twins/barlow_twins_4node_resnet \
+        config.MODEL.TRUNK.NAME=resnet config.MODEL.TRUNK.TRUNK_PARAMS.RESNETS.DEPTH=101 \
+        config.MODEL.TRUNK.TRUNK_PARAMS.RESNETS.WIDTH_MULTIPLIER=2
```

Comment:
we recently removed `TRUNK_PARAMS` from VISSL

Reply:
It seems that all docs in `ssl_approaches` are outdated.

Reply:
I don't really know how to update it. The docs are mainly a copy/paste of the `sim_clr` one

Reply:
hmmm, we did update our docs as well to reflect it. On quick search in VISSL master, https://github.com/facebookresearch/vissl/search?q=TRUNK_PARAMS I am not seeing the TRUNK_PARAMS in docs. 

What master are you at ? :)

Reply:
I can take care of updating the docs later in a follow-up PR :) we can proceed with this PR without doc updates! :)

Reply:
It should be good but I didn't patch the other docs.

Reply:
@prigoyal, 

See this file: https://github.com/facebookresearch/vissl/blob/master/docs/source/ssl_approaches/simclr.rst#training-different-model-architecture

Reply:
Github search is not reliable in my experience.

Reply:
Should I only remove `TRUNK_PARAMS`? I thought I was suppose to remove both `TRUNK` and `TRUNK_PARAMS`.

Reply:
OK I misunderstood you. I will add `TRUNK` back =P